libbpf-cargo/src/gen.rs

```diff
@@ -83,6 +107,74 @@ fn get_prog_name(prog: *const libbpf_sys::bpf_program) -> Result<String> {
     Ok(unsafe { CStr::from_ptr(name_ptr) }.to_str()?.to_string())
 }
 
+fn map_is_mmapable(map: *const libbpf_sys::bpf_map) -> bool {
+    let internal = unsafe { libbpf_sys::bpf_map__is_internal(map) };
+    let def = unsafe { libbpf_sys::bpf_map__def(map) };
+    let mmapable = unsafe { (*def).map_flags } & libbpf_sys::BPF_F_MMAPABLE;
+
+    internal && (mmapable > 0)
+}
+
+fn gen_skel_c_skel_constructor(
+    skel: &mut String,
+    object: *mut libbpf_sys::bpf_object,
+    name: &str,
+) -> Result<()> {
+    write!(
+        skel,
+        r#"
+        fn build_skel_config() -> libbpf_rs::Result<libbpf_rs::skeleton::ObjectSkeletonConfig<'static>>
+        {{
+            let mut builder = libbpf_rs::skeleton::ObjectSkeletonConfigBuilder::new(DATA);
+            builder
+                .name("{name}_bpf")
+        "#,
+        name = name
+    )?;
+
+    for map in MapIter::new(object) {
+        let raw_name = get_raw_map_name(map)?;
+        let mmaped = if map_is_mmapable(map) {
+            "true"
+        } else {
+            "false"
+        };
+
+        write!(
+            skel,
+            r#"
+            .map("{raw_name}", {mmaped})
+            "#,
+            raw_name = raw_name,
+            mmaped = mmaped,
```

Comment:
just curious, what write!() will write if you pass `bool` directly without "false"/"true" conversion? It capitalizes "True"/"False"?

Reply:
Compile error: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=06300b1a9b847065c1126116f4e7b92f

Reply:
I think it's supposed to work: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f26a3b1fd6c45d3a627280b579882a08 See https://doc.rust-lang.org/beta/std/macro.write.html

Reply:
specifically for bool, it seems like it does what I expected it to: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=621f1d7191930c55efb2438adbb207ea

Reply:
Interesting. A magic trait for everything...